src/components/utils/src/sql_qt_wrapper/sql_query.cc

```diff
@@ -0,0 +1,159 @@
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * Neither the name of the Ford Motor Company nor the names of its contributors
+ * may be used to endorse or promote products derived from this software
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "sql_qt_wrapper/sql_query.h"
+
+#include <QSqlDatabase>
+#include <QSqlError>
+#include <QString>
+#include <QStringList>
+#include <QVariant>
+
+#include <cassert>
+
+#include "sql_qt_wrapper/sql_database.h"
+
+namespace utils {
+namespace dbms {
+
+SQLQuery::SQLQuery(SQLDatabase* db)
+  : query_(static_cast<QSqlDatabase>(*db)) {
+}
+
+SQLQuery::~SQLQuery() {
+}
+
+bool SQLQuery::Prepare(const std::string& query) {
+  return query_.prepare(query.c_str());
+}
+
+bool SQLQuery::Exec() {
+  return query_.exec();
+}
+
+bool SQLQuery::Next() {
+  return query_.next();
+}
+
+bool SQLQuery::Reset() {
+  return true;
+}
+
+void SQLQuery::Finalize() {
+  query_.finish();
+}
+
+bool SQLQuery::Exec(const std::string& query) {
+  QString qstr = QString::fromStdString(query);
+  // QSqlQuery is unable to process several statements
+  // in one string, so need to split queries and execute them one by one
+  QStringList list = qstr.split(";", QString::SkipEmptyParts);
+  foreach (QString q, list) {
+    if (!query_.exec(q))
+      return false;
+  }
+  return true;
+}
+
+bool SQLQuery::GetBoolean(int pos) const {
+  const QVariant val = query_.value(pos);
+  return val.toBool();
+}
+
+int SQLQuery::GetInteger(int pos) const {
+  const QVariant val = query_.value(pos);
+  return val.toInt();
+}
+
+uint32_t SQLQuery::GetUInteger(int pos) const {
+  const QVariant val = query_.value(pos);
+  return val.toUInt();
+}
+
+int64_t SQLQuery::GetLongInt(int pos) const {
+  const QVariant val = query_.value(pos);
+  return val.toULongLong();
+}
+
+double SQLQuery::GetDouble(int pos) const {
+  const QVariant val = query_.value(pos);
```

Comment:
Actuallu all logging on the upper level. I just tried to keep  consistency with other implementations for `SQL`